modules/apps/commerce/commerce-product-test/src/testFunctional/tests/SFCommerceUnitsOfMeasure2.testcase

```diff
@@ -14,6 +14,635 @@ definition {
+			JSONLayoutpagetemplate.addDisplayPageTemplateEntry(
+				contentType = "Product",
+				displayPageTemplateEntryName = "Test Commerce Product Display Page Template",
+				groupName = "Minium");
+		}
+
+		task ("And an Option Selector fragment is deployed") {
+			DisplayPageTemplatesAdmin.openDisplayPagesAdmin(siteURLKey = "minium");
+
+			DisplayPageTemplatesAdmin.gotoDisplayPage(displayPageName = "Test Commerce Product Display Page Template");
+
+			CommerceDisplayPageTemplates.addFragment(
+				collectionName = "Product",
+				fragmentName = "Add to Cart",
+				index = 1);
+
+			CommerceDisplayPageTemplates.addFragment(
+				collectionName = "Product",
+				fragmentName = "Price",
+				index = 2,
+				targetFragmentName = "Add to Cart");
+
+			CommerceDisplayPageTemplates.addFragment(
+				collectionName = "Product",
+				fragmentName = "Availability",
+				index = 2,
+				targetFragmentName = "Price");
+		}
+
+		task ("And an account with a buyer user is created") {
+			CommerceEntry.addAccountEntryUser(
+				accountName = "Commerce Account",
+				accountType = "Business",
+				agreeToTermsAndAnswerReminderQuery = "true",
+				createNewAccount = "true",
+				requireReset = "false",
+				userEmailAddress = "buyer@liferay.com",
+				userFirstName = "Buyer",
+				userLastName = "User",
+				userRole = "Buyer",
+				userScreenName = "buyeruser",
+				userSiteMembership = "Minium");
+		}
+
+		task ("2 UOMs are created to MIN55861 with 0.6 as incremental order quantity") {
+			CommerceJSONSKUsAPI._addCommerceUOM(
+				basePrice = 20,
+				incrementalOrderQuantity = "0.6",
+				productSKU = "MIN55861",
+				uomKey = "UOM1KEY",
+				uomName = "UOM1");
+
+			CommerceJSONSKUsAPI._addCommerceUOM(
+				basePrice = 20,
+				incrementalOrderQuantity = "0.6",
+				productSKU = "MIN55861",
+				promoPrice = 15,
+				uomKey = "UOM2KEY",
+				uomName = "UOM2");
+		}
+
+		task ("And from the MIN55861 product configuration, multiple quantity is set as 0.6") {
+			CommerceJSONProductsAPI._patchCommerceProductConfiguration(
+				allowBackOrder = "true",
+				multipleOrderQuantity = "0.6",
+				productName = "U-Joint");
+		}
+
+		task ("When the buyer user logs in ") {
+			User.logoutPG();
+
+			CommerceLogin.miniumLogin(
+				password = "test",
+				urlAppend = "web/minium",
+				userEmailAddress = "buyer@liferay.com");
+		}
+
+		task ("Then assert U-Joint product card shows the"View All Variants" button instead of the add to cart") {
+			CommerceNavigator.searchEntry(entryName = "U-Joint");
+
+			AssertElementPresent(
+				key_text = "View all variants",
+				locator1 = "Button#ANY");
+		}
+
+		task ("When the buyer navigates to Minium catalog → MIN55861 product details") {
+			CommerceAcceleratorsInitializer.gotoMiniumProductDetailsPage(productName = "U-Joint");
+		}
+
+		task ("And assert that the information shown on the page is related to the selected SKU/UOM(availability, stock quantity, list price, promo price)") {
+			FrontStore.assertProductDetails(
```

Comment:
Add checks for stock quantity(add a Dynamic Field showing Inventory) and Promo Price

Thanks!